// hest/test/server.js
/**
 * Created by Alexander Nuikin (nukisman@gmail.com) on 05.10.15.
 */

var Hapi = require('hapi');
var server = new Hapi.Server();
var Boom = require('boom')
var Joi = require('joi')

var pluginHandler = function (err) {
    if (err) throw err
}

function registerPlugin(name) {
    server.register(require(name), pluginHandler)
}

server.connection({
    host: 'localhost',
    port: Number(8080)
});

server.route({
    method: 'GET', path: '/chickens/{breed}',
    config: {
        validate: {
            params: {
                breed: Joi.string().invalid('123')
            }
        }
    },
    handler: function (request, reply) {
        reply('Chicken breeds ' + request.params.breed)
    }
});

server.route({
    method: 'GET', path: '/test/{type}/{id}',
    config: {
        validate: {
            params: {
                type: Joi.string().required(),
                id: Joi.number().integer().required()
            }
        }
    },
    handler: function (request, reply) {
        reply({
            id: request.params.id,
            type: request.params.type
        })
    }
});

module.exports = server
